src/app/pages/rpc-stx-transfer-stx/rpc-stx-transfer-stx.tsx

```diff
@@ -1,25 +1,74 @@
 import { stxTransferStx } from '@leather.io/rpc';
-import { isDefined } from '@leather.io/utils';
+import { generateStacksUnsignedTransaction } from '@leather.io/stacks';
+import { StxAvatarIcon } from '@leather.io/ui';
+import { isString } from '@leather.io/utils';
 
-import { useLegacyRequestBroadcastTransaction } from '@app/common/legacy-requests';
-import { StacksHighFeeWarningContainer } from '@app/features/stacks-high-fee-warning/stacks-high-fee-warning-container';
-import { StacksTransactionSigner } from '@app/features/stacks-transaction-request/stacks-transaction-signer';
-import { useBreakOnNonCompliantEntity } from '@app/query/common/compliance-checker/compliance-checker.query';
+import { useConvertCryptoCurrencyToFiatAmount } from '@app/common/hooks/use-convert-to-fiat-amount';
+import { AccountStacksAddress } from '@app/components/account/account-stacks-address';
+import { TransactionRecipientsLayout } from '@app/components/rpc-transaction-request/transaction-recipients.layout';
+import { FeeEditor } from '@app/features/fee-editor/fee-editor';
+import { useFeeEditorContext } from '@app/features/fee-editor/fee-editor.context';
+import { NonceEditor } from '@app/features/nonce-editor/nonce-editor';
+import { useNonceEditorContext } from '@app/features/nonce-editor/nonce-editor.context';
+import { useRpcBroadcastStacksTransaction } from '@app/features/rpc-transaction-request/hooks/use-rpc-broadcast-stacks-transaction';
+import { RpcTransactionRequestLayout } from '@app/features/rpc-transaction-request/rpc-transaction-request.layout';
+import { SwitchAccountTrigger } from '@app/features/rpc-transaction-request/switch-account-trigger/switch-account-trigger';
+import { TransactionActionsWithSpend } from '@app/features/rpc-transaction-request/transaction-actions-with-spend';
+
+import { useRpcStxTransferStxContext } from './rpc-stx-transfer-stx.context';
 
 export function RpcStxTransferStx() {
-  const { onSignStacksTransaction, stacksTransaction, txPayload, txSender } =
-    useLegacyRequestBroadcastTransaction(stxTransferStx.method);
-  const recipient = 'recipient' in txPayload ? txPayload.recipient : '';
+  const { isLoading, onUserActivatesSwitchAccount, txOptions } = useRpcStxTransferStxContext();
+  const { availableBalance, isLoadingFees, marketData, onUserActivatesFeeEditor, selectedFee } =
+    useFeeEditorContext();
+  const { nonce, onUserActivatesNonceEditor } = useNonceEditorContext();
+  const broadcastTransaction = useRpcBroadcastStacksTransaction(stxTransferStx.method);
+  const convertToFiatAmount = useConvertCryptoCurrencyToFiatAmount('STX');
 
-  useBreakOnNonCompliantEntity([txSender, recipient].filter(isDefined));
+  async function onApproveTransaction() {
+    const unsignedTx = await generateStacksUnsignedTransaction(txOptions);
+    if (selectedFee.txFee) unsignedTx.setFee(selectedFee.txFee.amount.toNumber());
+    unsignedTx.setNonce(nonce);
+    await broadcastTransaction(unsignedTx);
+  }
```

Comment:
I like how here we very deterministically take values and create a new tx (rather than just pulling from the most recently generated state).

I wouldn't expect that `broadcastTransaction` would take care of signing the tx, though.